LEGO1/mxdsobject.cpp

```diff
@@ -0,0 +1,53 @@
+#include "mxdsobject.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+// OFFSET: LEGO1 0x100bf6a0
+MxDSObject::MxDSObject()
+{
+  // The following code yields 100% matching assembly if m_unk24 is declared as (signed) short.
+  // However, in other areas m_unk24 (notably, ISLE.EXE) is treated as unsigned short.
+  // this->m_unk0c = 0;
+  // this->m_unk10 = 0;
+  // this->m_unk14 = 0;
+  // this->m_name = NULL;
+  // this->m_unk24 = -1;
+  // this->m_unk1c = -1;
+  // this->m_unk28 = 0;
+
+  // Assembly not matching but m_unk24 is unsigned short (probably correct)
+  this->m_unk0c = 0;
+  this->m_unk10 = 0;
+  this->m_unk14 = 0;
+  this->m_name = NULL;
+  this->m_unk24 = 0xFFFF;
```

Comment:
Interesting, does anything change if you try casting a signed short into it? (e.g. `(unsigned short) -1`)

Reply:
`-1` = `0xff...` = `~0`. Just in case someone had missed that last one, and wants to double the number of permutations. :-)

Reply:
> I guess theoretically it's also possible that ISLE.EXE and LEGO1.DLL use disparate versions of this structure, one where m_unk24 is declared unsigned (ISLE.EXE) and one where it is signed (LEGO1.DLL). This way the code I've been looking at so far would match up. Pretty unlikely though.

What are the chances that it is a compiler optimization?
Maybe they were casting it to unsigned every time it was referenced, so the compiler changed the type to remove the casts?

I'm just making wild guesses here

Reply:
This is a very hacky idea and almost certainly not how the original code did this, but have you tried using a union so the same value can be both signed and unsigned?

Reply:
Good point, I guess that's a fair temporary workaround until we have figured this out. Added it, works!

Reply:
Re. (read) signed/unsigned discrepancies (and truncation, in case that's also encountered), my first suspect would be an inline accessor doing the cast/truncation.